splink/linker.py

```diff
@@ -100,12 +101,52 @@ def __setitem__(self, key, value):
         super().__setitem__(key, value)
 
         logger.log(
-            1, f"Setting cache for template name {key}" f" with physical name {value}"
+            1, f"Setting cache for {key}" f" with physical name {value.physical_name}"
         )
 
     def invalidate_cache(self):
         self.data = dict()
 
+    def get_with_logging(self, key):
```

Comment:
If you put the logging code in the `__getitem__` then it gets repeatedly called e.g. if you iterate over the dictionary.  We only want to log when we explicitly want to retrieve an item from the cache